superdesk/publish/__init__.py

```diff
@@ -35,6 +36,11 @@
 def register_transmitter(transmitter_type, transmitter, errors):
     registered_transmitters[transmitter_type] = transmitter
     transmitter_errors[transmitter_type] = dict(errors)
+    registered_transmitters_list.append({
+        'type': transmitter_type,
+        'name': transmitter.NAME or transmitter_type,
```

Comment:
I would use `getattr(transmitter, 'NAME', transmitter_type)` just in case `NAME` does not exist as attr